src/main/java/io/kestra/plugin/langchain4j/retriever/GoogleCustomWebSearchContentRetriever.java

```diff
@@ -0,0 +1,55 @@
+package io.kestra.plugin.langchain4j.retriever;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import dev.langchain4j.rag.content.retriever.ContentRetriever;
+import dev.langchain4j.rag.content.retriever.WebSearchContentRetriever;
+import dev.langchain4j.web.search.WebSearchEngine;
+import dev.langchain4j.web.search.google.customsearch.GoogleCustomWebSearchEngine;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.langchain4j.domain.ContentRetrieverProvider;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@Plugin(beta = true)
+@JsonDeserialize
+@Schema(
+    title = "WebSearch content retriever for Google Custom Search"
+)
+public class GoogleCustomWebSearchContentRetriever extends ContentRetrieverProvider {
+    @Schema(title = "API Key")
+    @NotNull
+    private Property<String> csi;
+
+    @Schema(title = "API Key")
+    @NotNull
+    private Property<String> apiKey;
+
+    @Schema(title = "Maximum number of results to return")
+    @NotNull
+    @Builder.Default
+    private Property<Integer> maxResults = Property.ofValue(3);
```

Comment:
As we said another properties in this plugin, let's not put a default value here since in addition it's already not null